### PR TITLE
Update FunctionInfo with computed virtual address

### PR DESCRIPTION
**Pull request by vwbaker**
Bug: http://b/249242994

### CODE REVIEW

**Review by karupayun — APPROVED** [review]
LGTM! But Valeriy should also approves.

PS: Don't forget to merge it in the release branch.

File: src/ClientData/CaptureData.cpp [hunk 1]
```diff
@@ -144,6 +144,12 @@ void CaptureData::ComputeVirtualAddressOfInstrumentedFunctionsIfNecessary(
         module_data->ConvertFromOffsetInFileToVirtualAddress(instrumented_function.file_offset());
     instrumented_function.set_function_virtual_address(virtual_address);
     ++updated_function_count;
+
+    ORBIT_CHECK(scope_id_provider_);
+    auto scope_id = scope_id_provider_->FunctionIdToScopeId(function_id);
+    if (scope_id.has_value()) {
```
Comment (akopich) [inline src/ClientData/CaptureData.cpp:150]: NIT. Maybe define scope_id under `if (...)` ?

File: src/ClientData/ScopeIdProvider.cpp [hunk 2]
```diff
@@ -146,4 +146,11 @@ const FunctionInfo* NameEqualityScopeIdProvider::GetFunctionInfo(ScopeId scope_i
   return nullptr;
 }
 
+void NameEqualityScopeIdProvider::UpdateFunctionInfoAddress(ScopeId scope_id, uint64_t address) {
+  const auto it = scope_id_to_function_info_.find(scope_id);
+  if (it != scope_id_to_function_info_.end()) {
```
Comment (akopich) [inline src/ClientData/ScopeIdProvider.cpp:151]: Nit. `it` under `if(...)`?

File: src/ClientData/CaptureData.cpp [hunk 3]
```diff
@@ -144,6 +144,12 @@ void CaptureData::ComputeVirtualAddressOfInstrumentedFunctionsIfNecessary(
...
+      scope_id_provider_->UpdateFunctionInfoAddress(scope_id.value(), virtual_address);
+    }
```
Comment (akopich) [inline src/ClientData/CaptureData.cpp:152]: This chunk of code should probably belong to  ScopeIdProvider implementation. Wouldn't things look neater if `UpdateFunctionInfoAddress` took a single parameter of type `InstrumentedFunction`?

File: src/ClientData/CaptureData.cpp [hunk 4]
```diff
@@ -144,6 +144,12 @@ void CaptureData::ComputeVirtualAddressOfInstrumentedFunctionsIfNecessary(
         module_data->ConvertFromOffsetInFileToVirtualAddress(instrumented_function.file_offset());
     instrumented_function.set_function_virtual_address(virtual_address);
     ++updated_function_count;
```
Comment (akopich) [inline src/ClientData/CaptureData.cpp:146]: I think, the increment should still be at the end of the loop. Like we've undated the data in the CaptureData, in the ScopeIdProvider and finally do the increment.

Comment (akopich) [inline src/ClientData/CaptureData.cpp:146]: I think, the increment should still be at the end of the loop. Like we've undated the data in the CaptureData, in the ScopeIdProvider and finally do the increment.

**Review by akopich — APPROVED** [review]
Thanks for the quick fix. LGTM otherwise.

File: src/ClientData/CaptureData.cpp (hunk 4 above)
Comment (vwbaker) [reply src/ClientData/CaptureData.cpp]: done

File: src/ClientData/ScopeIdProvider.cpp (hunk 2 above)
Comment (vwbaker) [reply src/ClientData/ScopeIdProvider.cpp]: done

**Pull request merged**